spec/menu_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'menu'
+
+describe Menu do
+  let(:dish1) { double(:dish, :name => "Baked Potato", :price => 3.25) }
+  let(:dish2) { double(:dish, :name => "Spaghetti Bolognese", :price => 4.70) }
+  let(:dish3) { double(:dish, :name => "Ham Sandwich", :price => 2.65) }
+  let(:dish4) { double(:dish, :name => "Tartiflette", :price => 6.80) }
+  let(:test_menu) { described_class.new("Fake Ass Restaurant") }
+
+  describe '#initialize' do
+    it 'has an empty array of dishes' do
+      expect(test_menu.dishes).to eq []
+    end
```

Comment:
seems to be testing state rather than behaviour - I defo recommend checking out the practical on this topic
essentially it advises avoiding tests on internal implementation and focusing on desired behaviour. so here instead of checking that it is an array, you could test:
`expect(test_menu.dishes).to be_empty`

then if you want to change your implentation internally for whatever reason, say to use a hash, your test still works